src/any_agent/evaluation/tools.py

```diff
@@ -0,0 +1,101 @@
+import json
+from collections.abc import Callable
+from typing import Any
+
+from pydantic import BaseModel
+
+from any_agent.tracing.agent_trace import AgentTrace
+
+MAX_EVIDENCE_LENGTH: int = 50
+
+
+class AgentEvaluationTools:
```

Comment:
I don't see the functions as being part of a class. The trace is not really a state, but rather an input of each function, afaict, so it won't change across invocations. Putting the functions at the module level is imho simpler and cleaner.